src/index/updater.rs

```diff
@@ -130,7 +117,7 @@ impl Updater {
         self.commit(wtx, value_cache)?;
```

Comment:
I think there should be a commit test as part of this if block. Something like:
```
let shouldCommitForSavepoint = starting_height - self.height < 15 && (self.height + uncommitted) % 5 === 0;

if uncommitted == 5000 || shouldCommitForSavepoint {
  self.commit(wtx, value_cache)?;
  ```

Disclaimer: Terrible variable naming and the logic might be off by 1 😋

Reply:
Or maybe a helper function in the reorg file